driver/TrEE/TrEE/VariableService.c

```diff
@@ -653,17 +653,24 @@ Return Value:
         &TeecOperation,
         &ErrorOrigin);
 
+    *ResultSize = (UINT32)TeecOperation.params[2].value.a;
+    *AuthVarStatus = (UINT32)TeecOperation.params[2].value.b;
+

```

Comment:
Why did this move?

Reply:
The return value was expected to contain the size even during a failure (specifically EFI_BUFFER_TOO_SMALL). This allows the caller to reallocate a larger buffer to contain the result and try again.